src/components/onBoarding/controllers/onBoardingWelcome.controller.js

```diff
@@ -1,16 +1,16 @@
 (function (angular) {
     'use strict';
     angular.module('znk.infra-web-app.onBoarding').controller('OnBoardingWelcomesController', ['userProfile', 'OnBoardingService', '$state', 'znkAnalyticsSrv',
-        function(userProfile, OnBoardingService, $state, znkAnalyticsSrv) {
+        function (userProfile, OnBoardingService, $state, znkAnalyticsSrv) {
 
             var onBoardingSettings = OnBoardingService.getOnBoardingSettings();
             this.username = userProfile.nickname || '';
 
             this.nextStep = function () {
                 var nextStep;
                 var nextState;
-                znkAnalyticsSrv.eventTrack({ eventName: 'onBoardingWelcomeStep' });
-                if (onBoardingSettings.showSchoolStep) {
+                znkAnalyticsSrv.eventTrack({eventName: 'onBoardingWelcomeStep'});
+                if (onBoardingSettings && onBoardingSettings.showSchoolStep) {
```

Comment:
we need to make sure that the OnBoardingService.getOnBoardingSettings() never returns undefined